uitools/toolkitcpp/import/Esri/ArcGISRuntime/Toolkit/ArcGISUserCredentialsView.qml

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+ *  Copyright 2012-2024 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+import Esri.ArcGISRuntime.Toolkit.Controller
+
+import QtQuick
+import QtQuick.Controls
+import QtQuick.Layouts
+
+Dialog {
+    id: userCredentialsView
+
+    property ArcGISAuthenticationController controller: null
+
+    title: qsTr("Authentication required")
+
+    closePolicy: Popup.NoAutoClose
+
+    header: Label {
+        text: `<h1>${title}</h1>`
+        horizontalAlignment: Text.AlignHCenter
+        padding: 5
+    }
+
+    footer: DialogButtonBox {
+        Button {
+            text: qsTr("Skip")
+            DialogButtonBox.buttonRole: DialogButtonBox.RejectRole
+        }
+
+        Button {
+            text: qsTr("Continue")
+            enabled: hasValidInput()
+            DialogButtonBox.buttonRole: DialogButtonBox.AcceptRole
+        }
+
+        onAccepted: {
+            acceptWithCurrentUsernameAndPassword();
```

Comment:
Which `accept` signal would this be responding to? An implicit signal emitted by the `DialogButtonBox`? I'm just confused because I see that this function also emits the `accept` signal.

Reply:
It's reacting to `DialogButtonBox` since that's the component it's within, and there is no `onAccepted` in the outer scope for the `Dialog`. It does call acceptWithCurrentUsernameAndPassword, which will call `accept` on the Dialog (coming soon in a commit), which is how it dismisses itself once credentials are supplied.

Also, this is identical to the existing [UserCredentialsView](https://github.com/Esri/arcgis-maps-sdk-toolkit-qt/blob/main/uitools/toolkitcpp/import/Esri/ArcGISRuntime/Toolkit/UserCredentialsView.qml#L50-L52).

Reply:
Okay I understand now, thanks for the explanation. So the eventual `accept` that comes from `acceptWithCurrentUsernameAndPassword` closes the `userCredentialsView` Dialog.